corehq/apps/es/filters.py

```diff
@@ -26,6 +26,10 @@ def prefix(field, value):
     return {"prefix": {field: value}}
 
 
+def wildcard(field, value):
+    return {"wildcard": {field: value}}
+
+
```

Comment:
From my understanding this is to do with the fact that the `user_data_es.value` property has an `analyzed` index. If it is an empty string, then there is nothing to analyze and so the empty string doesn't get picked up in the query. This is why doing something like:
```
queries.nested(
    USER_DATA_PATH,
    queries.filtered(
        queries.match_all(),
        filters.AND(
            filters.term('user_data_es.key', location_prop_name),
            filters.term('user_data_es.value', '')
        )
    )
)
```
won't return any results. Changing the index property to `not_analyzed` would cause a reindex, so instead I chose a solution to flip the search around and make the query exclude all documents that do have a value for this property. This is where the wildcard search comes in, as I use it to query documents that contain any value for the `user_data_es.value` property.

Reply:
> won't return any results

My understanding of ES is limited, so,
Can you confirm that you tested this or you already confident of this limitation?

Is there any performance implication of using wildcard?

Reply:
i am confident in this limitation. I have done a bit of trial-and-error attempting to do something similar to `CaseSearchES` with no positive results. After some research, I came to the above conclusion.

There is a performance hit when using wildcard, but the same performance impact is experienced when using the prefix filter, which we already support in `CaseSearchES`. Given this, and that there are far fewer mobile workers than cases this solution should be fine if we are able to use a similar approach for `CaseSearchES`.

Reply:
Thanks